src/mame/machine/k573mcr.cpp

```diff
@@ -31,4 +473,16 @@ const tiny_rom_entry *k573mcr_device::device_rom_region() const
 	return ROM_NAME( k573mcr );
 }
 
+
+INPUT_PORTS_START( k573mcr_meta_controls )
+	PORT_START("META")
+	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_SERVICE1) PORT_TOGGLE PORT_NAME("Insert/Eject Memory Card 1")
+	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_SERVICE2) PORT_TOGGLE PORT_NAME("Insert/Eject Memory Card 2")
+INPUT_PORTS_END
```

Comment:
I implemented meta controls for the memory card reader to simulate the action of inserting and ejecting the memory card slot. Is this ok? 

The intent was to make interactions like ending a credit in DDR smoother when it tells you to eject your card. Instead of waiting for the counter to tick down, it's possible to press the toggle button to eject your card and then press the button again to insert it, like you would on a real machine.

Reply:
Well, the usual way of removing/inserting media devices is using the file manager menu.  It becomes impractical when the media devices are slot devices that can instantiate different objects depending on what you insert into them because those require a hard reset to change.  However, as far as I can tell, the PlayStation memory isn’t one of these - the user can pull up the menu and take the memory card out.  It doesn’t require a special input.

Reply:
You are correct, the Playstation memory cards do not require a hard reset to change. My reasoning for this was that I think it's less practical to open the menu every time to eject and reopen the memory card file. Especially so if you are playing on something like a MAME cabinet and don't have as easy access to menus.

For a normal Playstation game you can reasonably expect that the player will generally keep the memory card in the slot. For the supported System 573 games, it's expected that the player will be removing and inserting memory cards after every credit and the games are coded as such. For example, unskippable timers until the player removes their card as a way to remind the player to take their card with them instead of leaving it in the machine.

It's not required but but I believe it's a keybind that smoothly replicates actions that the game expects you to perform. If this isn't convincing enough then I will remove it.

Reply:
Yeah, the UI could be better.  With the recent changes exposing media images to Lua, I think this can be handled in a more generic way using a plugin to allow binding keys to remove/reinsert last media.  That leaves the question of whether to leave the convenience hack here in the mean time or not.